Task/i 294/update card styles

rachelfranklin1 opened the pull request:
👋 This PR aims to solve issue #294. You can see the updated styling of the cards here: 

<img width="952" alt="Screenshot 2019-04-14 21 57 53" src="https://user-images.githubusercontent.com/26777813/56099083-645d8500-5f00-11e9-8e30-99d08af17737.png">

There were a couple of issues that need addressed:
1. there doesn't seem to be enough space for the full country name. It results in weird formatting of countries with a name longer than one word and there was an issue with the avatar being pushed over by the text, but that's easily resolved: 
<img width="581" alt="Screenshot 2019-04-14 21 56 15" src="https://user-images.githubusercontent.com/26777813/56099104-ab4b7a80-5f00-11e9-9f21-8ec5de1e47f0.png">
2. If a mentor has plenty of information, the elements move downwards, meaning that the card next to it has a different "layout" of elements: 
<img width="584" alt="Screenshot 2019-04-14 22 01 28" src="https://user-images.githubusercontent.com/26777813/56099146-1bf29700-5f01-11e9-8294-b4a65192552c.png">


I'd like to resolve this ☝🏻using CSS grid to keep the elements in a set space. But would like to do this in a separate PR.

moshfeu commented:
Thanks!
Will consult the ux team.

emmawedekind commented:
I'll respond ASAP. Also, please use the monospace font for the tech tags ;)

emmawedekind commented:
Hey I'm thinking about these two issues; also sidenote, can we vertically align the icons within the bottom social footer please? :) And when there are only two, please `justify-content: space-around` so that they're not on the left and right sides but somewhat centered with space around! Let me know if you need hi-fidelity mockups to explain!

rachelfranklin1 commented:
yep sure thing! Will do asap!

rachelfranklin1 commented:
Just a heads up that I've put some time aside to do this tomorrow 😻

moshfeu commented:
Thanks!

rachelfranklin1 commented:
I've added some padding to vertically align the icons and added `space-around` to align the footer icons. I think it looks perfect with three icons but can work some magic to only use `space-around` when there are less than three icons if you think it would be better?

<img width="916" alt="Screenshot 2019-04-19 16 07 32" src="https://user-images.githubusercontent.com/26777813/56430297-dbf33180-62bd-11e9-9b1c-d13c4b7f8323.png">

moshfeu commented:
LGTM :)

emmawedekind commented:
Love it!! One last nitpick; could we always keep the tags centered at the bottom? :) That way it doesn't look so intense when there isn't a quote!

<img width="699" alt="image" src="https://user-images.githubusercontent.com/7671983/56457255-191bfa00-6378-11e9-81c8-12267b450651.png">


Also, amazing work!

rachelfranklin1 commented:
Hey @emmawedekind I've updated the cards to use CSS grid and I also did a small refactor so that the card now renders 4 components: `<Header />`, `<MentorInfo />`, `<SkillsTags />`, & `<CardFooter />`.

It means that each component is a row of the CSS grid, and I think it is more readable :) 

You can see the updated styles here. Each row has a set height, so even if a description is not provided, it won't skew the alignment of the other rows: 
<img width="941" alt="Screenshot 2019-04-21 16 38 21" src="https://user-images.githubusercontent.com/26777813/56472269-1f7fa400-6454-11e9-9bc4-62c00d726805.png">

emmawedekind commented:
Looks amazing thank you!!
